CMakeLists.txt

```diff
@@ -30,7 +30,7 @@ add_subdirectory(third_party/antlr4/runtime/Cpp EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/UHDM)
 add_subdirectory(third_party/googletest EXCLUDE_FROM_ALL)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
```

Comment:
Do we need this for technical reasons or would only change it because UHDM also requires it ? (it might make sense to also look if we actually need C++14 there).

14 was an intermediate standard with not much changed since 11; 11 is desirable as many folks working in the hardware-related field often have super-old cent-os systems and sometimes struggle to get a newer compiler, so staying a few generations behind in the requirements opens it for more people.

So I am happy to accept a C++14 change if this is really needed to write more manageable code.

Reply:
So, looks like capnproto actually wants C++14, so in turn that means UHDM needs it. With that, I guess we can leave this here as minimum requirement.

Reply:
Ideally, I would go forward rather than backwards i.e. move to 17 instead of 11. As you said, v14 was an intermediate and things weren't locked down with the compilers. v17 is more polished. v11 lacks the platform support needed to keep the code clean of #if/#endif. Multiple cross platform features including filesystem, threads, and atomic moved out from non-experimental phase with this upgrade. Using experimental features is not ideal, especially for production code. But, that was the highest/easiest I could get to without raising eyebrows (because UHDM is already at 14) and still get some of the features needed to support other platforms.

> ... have super-old cent-os systems ...

I wouldn't disagree with that. I have had experienced this first hand. Given what Surelog/UHDM is (and wants to be), I contend that we have already made our choices.

* Using mostly C++ instead of primitive C
* Use of ANTLR for parsing (which happens to be Java based) instead of using flex/bison
* Use of python v3 instead of v2
* Choice of gcc7

Those choices speak for themselves. Those older systems you are talking about are already too old for the choices we have made. We have chosen what's current/active instead of what's the industry norm. I am not hearing an argument for holding back now.

I would also like to point out that Surelog is dependent on other third party modules (flatbuffers, googletest, and few others). Our hands might be tied if any one of those external dependencies decided to stop support for older standards or compilers. If anything we should be ahead of them not lagging in embracing newer technologies/standards. I surely do not wish to spend time reimplementing cross-platform filesystem IO or thread module.

Our focus should be on "completeness" of the parser and the technology, in general. If it can support all the old (and very old, in some cases) VHDL/Verilog code, that would be a bigger win then focusing our attention on which specific compiler or C++ standard to use. Nobody wants to reimplement what was implemented 30 years back. That stuff should just work because it has worked for three decades. You touch it and you break it.

All that said, if we decide to stick to v11, few class implementations would need to be changed. Surely, not the end of the world. A few coding guidelines to avoid sprinkling platform dependent code all over the source code would suffice.

> So I am happy to accept a C++14 change if this is really needed to write more manageable code.

This will be minimum required for filesystem support i.e. experimental filesystem support. Ideally, move up to C++17 to get non-experimental/production quality support from compilers.

@alaindargelas, @hzeller That's my penny! Will leave it to other stakeholders for final call.

Reply:
So personally, I'd go all the way to 17 as well for the std::filesystem reasons. And if there is a good time, it is now :)
I'll leave it to Alain to decide.